hamilton/driver.py

```diff
@@ -2010,13 +2031,25 @@ def build(self) -> Driver:
                 adapter=lifecycle_base.LifecycleAdapterSet(*adapter),
             )
 
+        if not self.module_overrides:  # if override on than this doesn't matter
```

Comment:
Hmm, interesting... This is actually a different type of module override -- those that are the same. Can you elaborate more on the case that it broke? Feels like an upstream thing... I'm not 100% sure I trust the `module in module_set` -- identity is a little weird/what happens if you make an update but keep an old reference around?

Reply:
yes, if  you do something like
```
.with_modules(module_A, module_A)
```
hamilton will just add it to the list of modules and treat them as two different ones. If you print the attribute, you get two pointers in the list
```
[<module 'module_A' from 'path to module A'>,<module 'module_A' from 'path to module A'>]
```
Then when you try to build the driver, because they are treated as separate you run into the issue that you have same-named functions and get an error that functions have to have different names.

The ``module_set`` tries to get create an ordered set out of the list we use for storing the module references.

Reply:
Yeah, I mean, maybe a separate PR? Or an issue with the proposed fix?